prov/util/src/util_attr.c

```diff
@@ -426,10 +426,13 @@ static int fi_resource_mgmt_level(enum fi_resource_mgmt rm_model)
 /* If a provider supports basic registration mode it should set the FI_MR_BASIC
  * mode bit in prov_mode. Support for FI_MR_SCALABLE is indicated by not setting
  * any of OFI_MR_BASIC_MAP bits. */
-int ofi_check_mr_mode(uint32_t api_version, uint32_t prov_mode,
+int ofi_check_mr_mode(uint32_t api_version, uint64_t caps, uint32_t prov_mode,
 		      uint32_t user_mode)
 {
 	if (FI_VERSION_LT(api_version, FI_VERSION(1, 5))) {
+		if (!(caps & FI_RMA))
```

Comment:
Should be:
```
if (!(caps & (FI_RMA | FI_ATOMIC)))
```
But I'm not sure we can discard FI_MR_ALLOCATED.  That applies in cases where local registration is needed.